src/pfs/src/pfsc.rs

```diff
@@ -100,14 +100,14 @@ impl<F: Read + Seek> Reader<F> {
             buf.fill(0);
         } else {
             // Read compressed.
-            let mut compressed = new_buffer(size as usize - 2);
+            let mut compressed = new_buffer(size as usize);
 
-            self.file.seek(SeekFrom::Start(offset + 2))?;
+            self.file.seek(SeekFrom::Start(offset))?;
             self.file.read_exact(&mut compressed)?;
 
             // Decompress.
-            let mut deflate = flate2::Decompress::new(false);
-            let status = match deflate.decompress(&compressed, buf, FlushDecompress::Finish) {
+            let mut deflate = ZlibDecoder::new(buf);
```

Comment:
`ZlibDecoder::new` accept [Read](https://doc.rust-lang.org/std/io/trait.Read.html) that provides Zlib data. That mean you should pass `compressed` as a slice of `u8` to it because a slice of `u8` implements [Read](https://doc.rust-lang.org/std/io/trait.Read.html#impl-Read-for-%26%5Bu8%5D).

`compressed` has a type of `Vec<u8>`. So you need to figure out how to get a slice of `u8` from a vec of `u8`. Good luck :)